Civi/Eck/API/Entity.php

```diff
@@ -90,13 +90,15 @@ public function onApi4EntityTypes(GenericHookEvent $event): void {
    */
   public static function afformEntityTypes(GenericHookEvent $e): void {
     foreach (\CRM_Eck_BAO_EckEntityType::getEntityTypes() as $entityType) {
-      $e->entities[$entityType['entity_name']] = [
-        'entity' => $entityType['entity_name'],
-        'label' => $entityType['label'],
-        'icon' => $entityType['icon'],
-        'type' => 'primary',
-        'defaults' => '{}',
-      ];
+      if ([] !== \CRM_Eck_BAO_EckEntityType::getSubTypes($entityType['name'])) {
```

Comment:
This could be simplified & commented like
```suggestion
      // Don't expose ECK entities without subtypes; they cannot be saved without a value for the required `subtype` field
      if (\CRM_Eck_BAO_EckEntityType::getSubTypes($entityType['name'])) {
```

Reply:
I recently switched to type-safe comparisons whenever possible, as that makes code more obvious for reading and also safer for static code analysis (our current *PHPStan* config would complain with a boolean comparison for an array method result).
As for the comment, I think I have to agree that subtypes being required doesn't seem obvious enough, so yeah, let's add that.